intern/frontend/src/index.jsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import { Cart } from './components/Cart/Cart'
+import { Shop } from './components/Shop/Shop'
+
+const App = () => {
+    const [products, setProducts] = React.useState([])
+    const [availableProducts, setAvailableProducts] = React.useState([])
+    const [inCartProducts, setInCartProducts] = React.useState([])
+
+    React.useEffect(() => {
+        fetch('http://localhost:8080/products')
+            .then((response) => response.json())
+            .then((products) => {
+                setProducts(products.products)
+            })
+    }, [])
+
+    React.useEffect(() => {
+        const initialAvailableProducts = products.reduce((acc, curr) => {
+            return { ...acc, [curr.id]: curr.stock }
+        }, {})
+        const initialInCartProducts = products.reduce((acc, curr) => {
+            return { ...acc, [curr.id]: 0 }
+        }, {})
+
+        setAvailableProducts(initialAvailableProducts)
+        setInCartProducts(initialInCartProducts)
+    }, [products])
+
+    const onAddToCart = (productId) => {
```

Comment:
I think we can add comments here too? To explain what we do and help candidates understand React? I feel like it may be hard as it is if they have no React experience at all :)